ssl/statem/statem_clnt.c

```diff
@@ -1897,6 +1897,7 @@ MSG_PROCESS_RETURN tls_process_server_certificate(SSL *s, PACKET *pkt)
                      ERR_R_MALLOC_FAILURE);
             goto err;
         }
+        X509_set0_libctx(x, s->ctx->libctx, s->ctx->propq);
```

Comment:
Probably we should do this straight after the d2i. We're already using `x` in the `tls_parse_all_extensions` call so it should have the libctx set by the time we do that.